File: FLIR/00_BasicSetup.py
#!/usr/bin/env python3

"""
Shows how to use ChunkData feature with QuickSpin API
"""

import PySpin


system = PySpin.System_GetInstance()

cam_list = system.GetCameras()

# Or by serial number
# cam_sn = 12345
# cam = cam_list.GetBySerial(cam_sn)
cam = cam_list.GetByIndex(0)
cam.Init()

print('Fancy processing')

cam.DeInit()
del cam
cam_list.Clear()
system.ReleaseInstance()

File: FLIR/00_BasicSetup.cpp
/*
 * Shows how to acquire an image.
 *
 * */


#include "Spinnaker.h"
#include <iostream>

using namespace Spinnaker;
using namespace Spinnaker::GenApi;
using namespace Spinnaker::GenICam;


int main(int /*argc*/, char** /*argv*/)
{
    std::cout << "FLIR: Acquisition\n";
    auto system = System::GetInstance();
    auto version = system->GetLibraryVersion();
    auto cam_list = system->GetCameras();

    std::cout << "Spinnaker Version: " << version.major << "."
              << version.minor << "." << version.type << "\n";

    if (cam_list.GetSize() == 0) {
        std::cerr << "No cameras found\n";
        cam_list.Clear();
        system->ReleaseInstance();
        return -1;
    } else {
        std::cout << cam_list.GetSize() << " Cameras Found\n";
    }
    auto cam = cam_list.GetByIndex(0);
    cam->Init();
    std::cout << "Fancy processing\n";
    cam->DeInit();

    // Release reference to the camera
    cam = nullptr;
    cam_list.Clear();
    system->ReleaseInstance();
    return 0;
}

